packages/time-picker/src/TimePicker/utils/filterTimeOptions.ts

```diff
@@ -0,0 +1,14 @@
+import { DateSegment } from "@react-stately/datepicker"
+import { TimeOption } from "../types"
+import { generateRegexFromInput } from "./generateRegexFromInput"
+
+export const generateFilteredTimeOptions = (
+  allOptions: Record<string, TimeOption>,
+  segments: DateSegment[]
+) =>
+  Object.keys(allOptions).reduce((filteredOptions, optionKey) => {
+    if (generateRegexFromInput(segments).test(optionKey)) {
```

Comment:
`segments` isn’t changing here but we are throwing away the result of `generateRegexFromInput` in each iteration of this loop. Perhaps perform `generateRegexFromInput` once outside the loop?